### PR TITLE
fix(task-analyzer): add app name normalization and typo correction

### LINKED ISSUE #149: [Bug]: Misspelled application names in task analyzer
### What happened

Task analyzer occasionally selects irrelevant app names leading to failure in task completion
Misspelling venmo as venom in Appworld is a typical example

### How to reproduce

Execute 5 runs of one of venmo tasks in appworld with gpt-oss-120b.
A good example is:
**024c982_1**	Request $13 publicly on Venmo from my friend, Stacy, with a note, "For yesterday's meal".

### Environment

**CUGA version:** Cuga-lite with Task analyzer, task decomposition and plan controller

### Logs, screenshots, or config

_No response_

### Checklist

- [x] I searched existing issues and this doesn’t look like a duplicate

### PR DESCRIPTION
**Pull request by offerakrabi**
## Bug fix

Fixes #149

### Summary

This PR adds robust app name normalization and typo correction to the TaskAnalyzer to handle cases where LLM-selected app names don't exactly match the available app names in the registry.

**Changes:**
- Added `resolve_relevant_apps()` method for robust app name matching with:
  - Case-insensitive exact matching with whitespace trimming
  - Fuzzy matching using SequenceMatcher for typo correction (0.8 similarity threshold)
  - Length delta guard (max 2 chars difference) to prevent false matches
  - Deduplication logic to handle equivalent app name variations
- Added validation to skip unresolved apps after matching with warning logs
- Imported difflib.SequenceMatcher for similarity scoring
- Added comprehensive unit tests (11 test cases) covering all matching scenarios

**Impact:**
- Prevents crashes when forced_apps contain typos or case variations
- Improves robustness of app matching in hybrid and API modes
- Maintains backward compatibility with existing behavior

### Testing
- [x] Verified fix locally; tests pass
- [x] Added 11 comprehensive unit tests covering all edge cases
- [x] Tests include: case-insensitive matching, whitespace trimming, empty/non-string handling, deduplication, similarity thresholds, length guards, forced apps typo correction, hybrid mode, and single app fast path

## Summary by CodeRabbit

* **New Features**
  * Improved app-matching: case-insensitive normalization, automatic correction of common typos, deduplication, and safer handling of empty/invalid entries to ensure only known apps are returned.

* **Bug Fixes**
  * Prevents producing outputs with unresolved or missing app definitions; ambiguous or unmatchable names are dropped with warnings.

* **Tests**
  * Added unit tests covering normalization, typo correction, ambiguity handling, deduplication, and mixed/forced input scenarios.

### CODE REVIEW

File: src/cuga/backend/cuga_graph/nodes/task_decomposition_planning/analyze_task.py [hunk 1]
```diff
@@ -45,6 +46,62 @@ def find_by_attribute(items: List[BaseModel], attr_name: str, attr_value) -> Opt
         except StopIteration:
             return None
 
+    @staticmethod
+    def resolve_relevant_apps(
```
Comment (sami-marreed) [inline src/cuga/backend/cuga_graph/nodes/task_decomposition_planning/analyze_task.py:50]: @offerakrabi please book a meeting on this fix and show it to us

**Review by nir-mashkif — APPROVED** [review]
approve after review with Nir and Sergey

**Review by sami-marreed — APPROVED** [review]
